Src/CSharpier.Cli/Options/ConfigurationFileOptions.cs

```diff
@@ -51,13 +38,39 @@ ConfigurationFileOptions configurationFileOptions
         };
     }
 
-    public static ConfigurationFileOptions FindForDirectory(
-        string baseDirectoryPath,
+    /// <summary>Finds all configs above the given directory as well as within the subtree of this directory</summary>
+    internal static List<CSharpierConfigData> FindForDirectoryName(
+        string directoryName,
         IFileSystem fileSystem,
-        ILogger? logger = null
+        ILogger logger
     )
     {
-        var directoryInfo = fileSystem.DirectoryInfo.FromDirectoryName(baseDirectoryPath);
+        var results = new List<CSharpierConfigData>();
+        var directoryInfo = fileSystem.DirectoryInfo.FromDirectoryName(directoryName);
+
+        var filesByDirectory = directoryInfo
+            .EnumerateFiles(".csharpierrc*", SearchOption.AllDirectories)
```

Comment:
So, to confirm: you're looking for all `.csharpierrc` files in subdirectories, but you're only going to apply them to files that are in that directory (and below), right?

Because it would be weird if someone created a `.csharpierrc` (or `.editorconfig`) for a specific project in a monorepo and suddenly files in its parent directories started formatting that way.

Reply:
> you're only going to apply them to files that are in that directory (and below), right?

Correct. I didn't test it, but it seemed safe to assume that it would be faster to first find all possible configuration files in the directory tree being formatted. Then when formatting a given file, figure out which configuration applies to it.